### DISCUSSION TITLE
Build Fails at `glib-sys v0.5.0`

### DISCUSSION THREAD

**Discussion by juristoeckli** [mum-rs/mum#131] [Q&A]
error: failed to run custom build command for `glib-sys v0.5.0`

Caused by:
  process didn't exit successfully: `/tmp/cargo-install3kZc2w/release/build/glib-sys-255260c710c5480b/build-script-build` (exit code: 1)
--- stdout
cargo:rerun-if-env-changed=GLIB_2.0_NO_PKG_CONFIG
cargo:rerun-if-env-changed=PKG_CONFIG
cargo:rerun-if-env-changed=GLIB_2.0_STATIC
cargo:rerun-if-env-changed=GLIB_2.0_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
cargo:rerun-if-env-changed=PKG_CONFIG_PATH_arm-unknown-linux-gnueabihf
cargo:rerun-if-env-changed=PKG_CONFIG_PATH_arm_unknown_linux_gnueabihf
cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
cargo:rerun-if-env-changed=PKG_CONFIG_PATH
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_arm-unknown-linux-gnueabihf
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_arm_unknown_linux_gnueabihf
cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_arm-unknown-linux-gnueabihf
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_arm_unknown_linux_gnueabihf
cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

--- stderr
`"pkg-config" "--libs" "--cflags" "glib-2.0" "glib-2.0 >= 2.32"` did not exit successfully: exit code: 1
--- stderr
Package glib-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `glib-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'glib-2.0' found
Package glib-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `glib-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'glib-2.0' found

warning: build failed, waiting for other jobs to finish...
error: failed to compile `mumd v0.3.0`, intermediate artifacts can be found at `/tmp/cargo-install3kZc2w`

Caused by:
  build failed
pi@raspberrypi:~/mum $

**Comment by sornas** [comment]
The only part requiring glib should be the notifications through libnotify. You have two options:

1) Install glib in some way. I can't test it myself but if you're running on a Raspberry Pi then `libglib2.0-dev` might be what you're looking for. (`sudo apt search libglib` in case you want to take a look for yourself.) However, libnotify should have already installed glib as a dependency so check if you have that as well (`libnotify-dev` I believe).

2) Build without notification support. The README talks a bit about how you'd do this under "Optional dependencies". This becomes much easier this Thursday when 1.51 drops (and I'll update the README to match this).
